linked_list/linked_list.py

```diff
@@ -86,11 +163,22 @@ def visit(self):
 
     # method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def reverse(self):

```

Comment:
✨ Very nice, however you aren't creating a call stack proportional to the input or creating a new input so space complexity would be O(1) here.